src/Adapter/StockManager.php

```diff
@@ -139,24 +147,37 @@ private function updateReservedProductQuantity($shopId, $errorState, $cancellati
                 sa.id_product_attribute = od.product_attribute_id
                 GROUP BY sa.id_product, sa.id_product_attribute
             )
+            WHERE sa.id_shop = :shop_id
         ';
 
-        $updateReservedQuantityQuery = str_replace(
-            array(
-                '{table_prefix}',
-                ':shop_id',
-                ':error_state',
-                ':cancellation_state',
-            ),
-            array(
-                _DB_PREFIX_,
-                (int)$shopId,
-                (int)$errorState,
-                (int)$cancellationState,
-            ),
-            $updateReservedQuantityQuery
+        $arrayFrom = array(
+            '{table_prefix}',
+            ':shop_id',
+            ':error_state',
+            ':cancellation_state',
+        );
+
+        $arrayTo = array(
+            _DB_PREFIX_,
+            (int)$shopId,
+            (int)$errorState,
+            (int)$cancellationState,
         );
 
+        if ($idProduct) {
+            $updateReservedQuantityQuery .= ' AND sa.id_product = :product_id';
+            $arrayFrom[] = ':product_id';
+            $arrayTo[] = (int)$idProduct;
+        }
+
+        if ($idOrder) {
+            $updateReservedQuantityQuery .= ' AND sa.id_product IN (SELECT product_id FROM {table_prefix}order_detail WHERE id_order = :order_id)';
+            $arrayFrom[] = ':order_id';
+            $arrayTo[] = (int)$idOrder;
+        }
+
+        $updateReservedQuantityQuery = str_replace($arrayFrom, $arrayTo, $updateReservedQuantityQuery);
```

Comment:
It's better to use `strtr` instead of `str_replace`. Not only its clearer because it takes a single array of "search" => "replace" pairs, but it also prevents [this kind of problems](http://php.net/manual/en/function.strtr.php#117169)